File: algorithms/insertion_sort/javascript/insertion_sort.js
function insertionSort(arr){
    for(let i=1; i<arr.length; i++){
        let key = arr[i];
        let j = i-1;
        while(j>=0 && arr[j]>key){
            arr[j+1] =arr[j];
            j--;
        }
        arr[j+1] = key;
    }
    console.log(arr);
}
insertionSort([1,3,4,2,0,8]);

File: algorithms/insertion_sort/haskell/insertion_sort.hs
main :: IO()
main = do
  let sorted = insertionSort [6, 5, 3, 1, 8, 7, 2, 4] :: [Integer]
  print sorted

insert :: Ord a => a -> [a] -> [a]
insert x [] = [x]
insert x (y:ys) | x < y     = x:y:ys
                | otherwise = y:(insert x ys)

insertionSort :: Ord a => [a] -> [a]
insertionSort [] = []
insertionSort (x:xs) = insert x (insertionSort xs)